src/town_cmd.cpp

```diff
@@ -1136,9 +1137,52 @@ static bool GrowTownWithBridge(const Town *t, const TileIndex tile, const DiagDi
 		} while (IsValidTile(bridge_tile) && (IsWaterTile(bridge_tile) || IsPlainRailTile(bridge_tile) || (IsNormalRoadTile(bridge_tile) && GetDisallowedRoadDirections(bridge_tile) != DRD_NONE)));
 	}
 
-	/* no water tiles in between? */
+	/* Don't allow bridges with no span between end tiles. */
 	if (bridge_length == 1) return false;
 
+	/* Next we check the tile beyond the bridge to ensure the road can continue. First check if the tile is valid. */
+	if (IsValidTile(bridge_tile)) {
```

Comment:
It shouldn't check for `bridge_tile + delta` validity instead?

Reply:
Yes, thanks.